python/pyarrow/tests/test_pandas.py

```diff
@@ -107,10 +107,6 @@ def _check_series_roundtrip(s, type_=None, expected_pa_type=None):
         assert arr.type == expected_pa_type
 
     result = pd.Series(arr.to_pandas(), name=s.name)
-    if pa.types.is_timestamp(arr.type) and arr.type.tz is not None:
-        result = (result.dt.tz_localize('utc')
-                  .dt.tz_convert(arr.type.tz))
```

Comment:
The series roundtrip with timezone was actually already tested in this file, but this fixup of the result was masking the issue. So before, it was expected that Array.to_pandas would loose the timezone (maybe because this method could also return numpy arrays in some cases). But I don't see a reason to not keep it (certainly now it always returns Series, and now Column is gone, which retained the timezone)